scipy/stats/tests/test_discrete_distns.py

```diff
@@ -112,3 +114,81 @@ def test_skellam_gh11474():
                     0.5044605891382528, 0.5019947363350450, 0.5019848365953181,
                     0.5019750827993392, 0.5019466621805060, 0.5018209330219539]
     assert_allclose(cdf, cdf_expected)
+
+
+class TestZipfian:
+    def test_zipfian_asymptotic(self):
+        # test limiting case that zipfian(a, n) -> zipf(a) as n-> oo
+        a = 6.5
+        N = 10000000
+        k = np.arange(1, 21)
+        assert_allclose(zipfian.pmf(k, a, N), zipf.pmf(k, a))
+        assert_allclose(zipfian.cdf(k, a, N), zipf.cdf(k, a))
+        assert_allclose(zipfian.sf(k, a, N), zipf.sf(k, a))
+        assert_allclose(zipfian.stats(a, N, moments='msvk'),
+                        zipf.stats(a, moments='msvk'))
+
+    def test_zipfian_continuity(self):
+        # test that zipfian(0.999999, n) ~ zipfian(1.000001, n)
+        # (a = 1 switches between methods of calculating harmonic sum)
+        alt1, agt1 = 0.99999999, 1.00000001
+        N = 30
+        k = np.arange(1, N  + 1)
+        assert_allclose(zipfian.pmf(k, alt1, N), zipfian.pmf(k, agt1, N))
+        assert_allclose(zipfian.cdf(k, alt1, N), zipfian.cdf(k, agt1, N))
+        assert_allclose(zipfian.sf(k, alt1, N), zipfian.sf(k, agt1, N))
+        assert_allclose(zipfian.stats(alt1, N, moments='msvk'),
+                        zipfian.stats(agt1, N, moments='msvk'), rtol=2e-7)
+
+    def test_zipfian_R(self):
```

Comment:
I should remove this test against R because the R implementation is wrong. For example, run this code at [rdrr.io](https://rdrr.io/cran/VGAM/man/zipfUC.html).
```R
library(VGAM)
options(digits=16)
k <- 3
a <- 1.2589254117941673
n <- 12
pmf <- dzipf(k, N = n, shape = a)
cdf <- pzipf(k, N = n, shape = a)
print(pmf)
print(cdf)
```
For `n <= 12`, the two implementations match to machine precision.
Increase `n` to 13, and the error is ~0.001. You can tell R is wrong because if you set:
```R
k <- 0:14
```
and sum the pmf values, they don't add to 1.

Reply:
So I assume the goal of this test is to provide reasonable assurance that this SciPy implementation is doing a reasonable thing compared to another independent implementation, in this case R.  Is Wolfram's `ZipfDistribution` the correct distribution? If not, might be okay to omit this test if we're testing against bad values

FYI, appears that R [is using](https://github.com/cran/VGAM/blob/ad275cd8262668ccf792601c74c008f5014285d9/R/family.aunivariate.R#L4815) what you call below the "naive" method

Reply:
Yes, the idea is just to compare against an existing implementation.

>  Is Wolfram's ZipfDistribution the correct distribution?

No, that's the zeta distribution. The confusion between the two distributions (gh-11523) is why I created this PR in the first place, actually : )  So another thing I was thinking we might want to do is doc-deprecate `zipf` and give it an alias `zeta`.

Reply:
Nope, I don't think it's license compatible, it's GPL 2 or 3